path/basename_test.ts

```diff
@@ -4,33 +4,94 @@
 import { assertEquals } from "../testing/asserts.ts";
 import * as path from "./mod.ts";
 
+// Test suite from "GNU core utilities"
+// https://github.com/coreutils/coreutils/blob/master/tests/misc/basename.pl
+const COREUTILS_TESTSUITE = [
+  [["d/f"], "f"],
+  [["/d/f"], "f"],
+  [["d/f/"], "f"],
+  [["d/f//"], "f"],
+  [["f"], "f"],
+  [["/"], "/"],
+  [["///"], "/"],
+  [["///a///"], "a"],
+  [[""], ""],
+  [["aa", "a"], "a"],
+  [["a-a", "-a"], "a"],
+  [["f.s", ".s"], "f"],
+  [["fs", "s"], "f"],
+  [["fs", "fs"], "fs"],
+  [["fs/", "s"], "f"],
+  [["dir/file.suf", ".suf"], "file"],
+  [["fs", "x"], "fs"],
+  [["fs", ""], "fs"],
+  [["fs/", "s/"], "fs"],
+];
+
+const POSIX_TESTSUITE = [
+  [[""], ""],
+  [["/dir/basename.ext"], "basename.ext"],
+  [["/basename.ext"], "basename.ext"],
+  [["basename.ext"], "basename.ext"],
+  [["basename.ext/"], "basename.ext"],
+  [["basename.ext//"], "basename.ext"],
+  [["aaa/bbb", "/bbb"], "bbb"],
+  [["aaa/bbb", "a/bbb"], "bbb"],
+  [["aaa/bbb", "bbb"], "bbb"],
+  [["aaa/bbb//", "bbb"], "bbb"],
+  [["aaa/bbb", "bb"], "b"],
+  [["aaa/bbb", "b"], "bb"],
+  [["/aaa/bbb", "/bbb"], "bbb"],
+  [["/aaa/bbb", "a/bbb"], "bbb"],
+  [["/aaa/bbb", "bbb"], "bbb"],
+  [["/aaa/bbb//", "bbb"], "bbb"],
+  [["/aaa/bbb//", "a/bbb"], "bbb"],
+  [["/aaa/bbb", "bb"], "b"],
+  [["/aaa/bbb", "b"], "bb"],
+  [["/aaa/bbb"], "bbb"],
+  [["/aaa/"], "aaa"],
+  [["/aaa/b"], "b"],
+  [["/a/b"], "b"],
+  [["//a"], "a"],
+  [["///"], "/"],
+  [["///", "bbb"], "/"],
+  [["//", "bbb"], "/"],
+];
+
+const WIN32_TESTSUITE = [
+  [["\\dir\\basename.ext"], "basename.ext"],
+  [["\\basename.ext"], "basename.ext"],
+  [["basename.ext"], "basename.ext"],
+  [["basename.ext\\"], "basename.ext"],
+  [["basename.ext\\\\"], "basename.ext"],
+  [["foo"], "foo"],
+  [["aaa\\bbb", "\\bbb"], "bbb"],
+  [["aaa\\bbb", "a\\bbb"], "bbb"],
+  [["aaa\\bbb", "bbb"], "bbb"],
+  [["aaa\\bbb\\\\\\\\", "bbb"], "bbb"],
+  [["aaa\\bbb", "bb"], "b"],
+  [["aaa\\bbb", "b"], "bb"],
+  [["/aaa/bbb", "bb"], "b"],
+  [["C:"], ""],
+  [["C:."], "."],
+  [["C:\\"], ""],
```

Comment:
Because the result of `posix.basename('/')` is `/`, so i think the result of `win32.basename('C:\\')` is `\\`.

Reply:
Fixed